pkg/routeagent_driver/handlers/mtu/mtuhandler.go

```diff
@@ -0,0 +1,144 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mtu
+
+import (
+	"fmt"
+	"strings"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/event"
+	"github.com/submariner-io/submariner/pkg/iptables"
+)
+
+const (
+	postRoutingChain           = "POSTROUTING"
+	subMarinerPostRoutingChain = "SUBMARINER-POSTROUTING"
+	mangleTable                = "mangle"
+)
+
+type mtuHandler struct {
+	event.HandlerBase
+	ipt iptables.Interface
+}
+
+func NewMTUHandler() event.Handler {
+	return &mtuHandler{}
+}
+
+func (h *mtuHandler) GetNetworkPlugins() []string {
+	return []string{event.AnyNetworkPlugin}
+}
+
+func (h *mtuHandler) GetName() string {
+	return "MTU handler"
+}
+
+func (h *mtuHandler) Init() error {
+	var err error
+	h.ipt, err = iptables.New()
+	if err != nil {
+		return fmt.Errorf("error initializing iptables: %v", err)
+	}
+
+	if err := iptables.CreateChainIfNotExists(h.ipt, mangleTable, subMarinerPostRoutingChain); err != nil {
+		return fmt.Errorf("error creating iptables chain %s: %v", subMarinerPostRoutingChain, err)
+	}
+
+	forwardToSubMarinerPostRoutingChain := []string{"-j", subMarinerPostRoutingChain}
+
+	if err := iptables.PrependUnique(h.ipt, mangleTable, postRoutingChain, forwardToSubMarinerPostRoutingChain); err != nil {
+		return fmt.Errorf("error inserting iptables rule %q: %v",
+			strings.Join(forwardToSubMarinerPostRoutingChain, " "), err)
+	}
+
+	return nil
+}
+
+func (h *mtuHandler) RemoteEndpointCreated(endpoint *submV1.Endpoint) error {
+	subnets := extractSubnets(endpoint.Spec)
+	for _, subnet := range subnets {
+		err := h.addClampMssToPathMTU(subnet)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *mtuHandler) RemoteEndpointRemoved(endpoint *submV1.Endpoint) error {
+	subnets := extractSubnets(endpoint.Spec)
+	for _, subnet := range subnets {
+		err := h.removeClampMssToPathMTU(subnet)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *mtuHandler) addClampMssToPathMTU(subnets string) error {
+	ruleSpecSource := []string{"-s", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
```

Comment:
Normally it's a good practice to include a match on both the sourceCIDR and destinationCIDR to avoid any leaks. This is what we do in the route-agent rules programmed in the NAT table (SUBMARINER-POSTROUTING chain).

Since this is the first time we are adding these rules, personally I feel that it's good to go directly with ipsets (just like how we are using them in the new Globalnet 2.0). It would be pretty straightforward for this use-case.
 
1. Create two IPset chains with a specific name (say remoteCIDRs and localCIDRs)
2. Add an iptable rule in the mange table, that matches on the ipset source localCIDR and ipset destination remoteCIDR - perform the mss-clamp-action.
3. Add an iptable rule in the mangle table, that matches on the ipset source remoteCIDR and ipset destination localCIDR - perform the mss-clamp-action.
3. When a new remoteEndpoint is added, update the remoteCIDR ipset chain and when the remoteEndpoint is deleted remote the entry.

So irrespective of how many clusters are connected, we will only be using two iptable rules in the mangle table.
It will be very optimal and less error-prone.

CC: @mangelajo @nyechiel

Reply:
Yes, I like that, but may be @aswinsuryan may need some help with that. Can you give him some pointers? (I believe we do it in globalnet, right?)

Reply:
Yes, we are using it for GlobalEgressIPs with namespace/podSelectors here - https://github.com/submariner-io/submariner/blob/debfc47870e4e9239dbaa6035f3a38c0c1882ad3/pkg/globalnet/controllers/global_egressip_controller.go#L65

Feel free to reach out if you have any questions @aswinsuryan